### DISCUSSION TITLE
Render ViewComponent in Blade by classname?

### DISCUSSION THREAD

**Discussion by cao-minh-duc** [laravel/framework#33534] [General] [answered]
Is there any way to render ViewComponent in Blade like @spatie ViewComponent

Example: Directive @render(App\Http\View\MyComponent::class)

**Comment by driesvints** [comment, marked as answer]
No. You need Spatie's package for that.

**Reply by cao-minh-duc** [reply]
I'm trying a solution by making a directive 

`Blade::directive('render', function ($componentClassname,$data = '[]') {
            return "<?php \$component = app($componentClassname); \$view = \$component->resolveView(); \$data = array_merge(\$component->data(),$data); \$view->with(\$data); ?> {{ \$view }}";
        });`

Can I make a PR?
